libeos-parental-controls/app-filter.c

```diff
@@ -303,7 +303,7 @@ epc_app_filter_is_appinfo_allowed (EpcAppFilter *filter,
 
           for (gsize i = 0; old_flatpak_apps[i] != NULL; i++)
             {
-              gchar *old_flatpak_app = g_strstrip (old_flatpak_app);
```

Comment:
Hmm, interesting. Putting that code into `app-filter.c` gives me a warning. Undoing the `old_flatpak_app` fix doesn’t give me a warning. Putting the following code (i.e. wrapping your code in a loop) into `app-filter.c` also *doesn’t* give me a warning:
```
  for (gsize i = 0; i < 10; i++)
    {
      char *bar = g_strstrip (bar);
    }
```

So it looks like the compiler’s heuristics for detecting `-Wmaybe-uninitialized` get broken by loops.

I’m using GCC 8.2.1. The bug might be https://gcc.gnu.org/bugzilla/show_bug.cgi?id=54554, since I’m compiling at `-O0`; but looking at the list of ‘uninitialized’ bugs in the GCC Bugzilla, there’s enough there that it’s either been reported already, or reporting it would not get it fixed.

In any case, Meson does seem to be successfully passing the warning flags in to the compiler command.